dev/sbt-plugin/src/sbt-test/sbt-plugin/akka-persistence-typed-migration-scala/shopping-cart-api/src/main/scala/com/example/shoppingcart/api/ShoppingCartService.scala

```diff
@@ -0,0 +1,110 @@
+
+object ShoppingCartService  {
+  val TOPIC_NAME = "shopping-cart"
+}
+
+/**
+  * The ShoppingCart service interface.
+  * <p>
+  * This describes everything that Lagom needs to know about how to serve and
+  * consume the ShoppingCartService.
+  */
+trait ShoppingCartService extends Service {
+
+  /**
+    * Get a shopping cart.
+    *
+    * Example: curl http://localhost:9000/shoppingcart/123
+    */
+  def get(id: String): ServiceCall[NotUsed, String]
+
+  /**
+   * Get a shopping cart report (view model).
+   *
+   * Example: curl http://localhost:9000/shoppingcart/123/report
+   */
+  def getReport(id: String): ServiceCall[NotUsed, String]
+
+  /**
+    * Update an items quantity in the shopping cart.
+    *
+    * Example: curl -H "Content-Type: application/json" -X POST -d '{"productId": 456, "quantity": 2}' http://localhost:9000/shoppingcart/123
+    */
+  def updateItem(id: String, productId: String, qty: Int): ServiceCall[NotUsed, Done]
+
+  /**
+    * Checkout the shopping cart.
+    *
+    * Example: curl -X POST http://localhost:9000/shoppingcart/123/checkout
+    */
+  def checkout(id: String): ServiceCall[NotUsed, Done]
+
+  override final def descriptor = {
+    import Service._
+    // @formatter:off
+    named("shopping-cart")
+      .withCalls(
+        restCall(Method.GET, "/shoppingcart/:id", get _),
+        restCall(Method.GET, "/shoppingcart/:id/report", getReport _),
+        // for the RESTafarians, my formal apologies but the GET calls below do mutate date
+        // we just want an easy way to mutate data from a sbt scripted test, so no POST/PUT here
+        restCall(Method.GET, "/shoppingcart/:id/:productId/:num", updateItem _),
+        restCall(Method.GET, "/shoppingcart/:id/checkout", checkout _)
+      )
+      .withAutoAcl(true)
+    // @formatter:on
+  }
+}
+
+/**
+  * A shopping cart item.
+  *
+  * @param productId The ID of the product.
+  * @param quantity The quantity of the product.
+  */
+case class ShoppingCartItem(productId: String, quantity: Int)
+
+object ShoppingCartItem {
+  /**
+    * Format for converting the shopping cart item to and from JSON.
+    *
+    * This will be picked up by a Lagom implicit conversion from Play's JSON format to Lagom's message serializer.
+    */
+  implicit val format: Format[ShoppingCartItem] = Json.format
+}
+
+/**
+  * A shopping cart.
+  *
+  * @param id The id of the shopping cart.
+  * @param items The items in the shopping cart.
+  * @param checkedOut Whether the shopping cart has been checked out (submitted).
+  */
+case class ShoppingCart(id: String, items: Seq[ShoppingCartItem], checkedOut: Boolean)
+
+object ShoppingCart {
+
+  implicit val format: Format[ShoppingCart] = Json.format
+}
+
+
+/**
+ * A shopping cart report exposes information about a ShoppingCart.
+ */
+case class ShoppingCartReport(cartId: String, created: Boolean, checkedOut: Boolean)
+
+object ShoppingCartReport {
+  implicit val format: Format[ShoppingCartReport] = Json.format
+}
```

Comment:
Looking at the signatures of the methods in the API above I think none of this payload types is used.